### ISSUE TITLE
Tests crash when a Test Fixture has multiple parameterized tests.

### ISSUE THREAD

**Issue by DarkKilauea** [CodeDarigan/WAT#124]
Godot Version: v3.2.1.stable.official
WAT Version: master (37456f7e8cbdbafd9bf8c05d2a8b6fbaf2867e32) and v4.1.0

When one has multiple parameterized tests in the same Test Fixture, the second test that runs will crash due to the parameters not properly refreshing to reflect the new test.  This effect is also evident on the 4.1.0 version, though you need to have one more test than the number of parameters available in the first test that runs.

The crash occurs at `res://addons/WAT/test/parameters.gd:20 - at function:_update`.  The `_values` array still holds the data from the last test that ran, which is empty due to all the parameterized instances having already run.

I believe this can be fixed by replacing `if _keys.empty():` with`if _keys.empty() or _values.empty():` on line 10 of `res://addons/WAT/test/parameters.gd`.

Repo:
```
extends WATTest

func test_add():
	parameters([["addend", "augend", "result"], [2, 2, 4], [5, 3, 8], [7, 6, 13]])
	describe("When we add %s to %s we get %s" % [p.addend as String, p.augend as String, p.result as String])
	var expected = p.result
	var actual = calculator.add(p.addend, p.augend)
	asserts.is_equal(actual, expected)

func test_subtract():
	parameters([["addend", "augend", "result"], [2, 2, 0], [5, 3, 2], [7, 6, 1]])
	describe("When we subtract %s from %s we get %s" % [p.addend as String, p.augend as String, p.result as String])
	var expected = p.result
	var actual = calculator.sub(p.addend, p.augend)
	asserts.is_equal(actual, expected)

class calculator:
	static func add(a, b):
		return a + b;
	
	static func sub(a, b):
		return a - b;

```

**Comment by CodeDarigan** [comment]
Added this fix in master in d6d2902fb278b699e334f57983039f5bdfda6018. Tests pass though I'm still unsure if there is possibly something else going on so I'm going to keep this issue open for a while.

**Comment by CodeDarigan** [comment]
Closing this since no problems seem to have arisen since.
